basilisk/src/ast/ast.c

```
@@ -0,0 +1,1191 @@
+  return n;
+}
+
+AstRoot * ast_parse_file (FILE * fp, AstRoot * parent)
+{
+  char * buffer = NULL;
+  size_t len = 0, maxlen = 0;
+  int c;
+  while ((c = fgetc (fp)) != EOF) {
+    if (len >= maxlen) {
+      maxlen += 4096;
+      buffer = realloc (buffer, maxlen);      
```

Cppcheck (2.10-2)

[error] Common realloc mistake

(memleakOnRealloc)